tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -266,14 +258,15 @@ def _fit(self, X_train: pd.DataFrame, y_train: pd.Series,
     def save(self, path: str = None, verbose=True) -> str:
         if path is None:
             path = self.path
-        model_path = os.path.join(path, self.model_file_name)
         text_nn_path = os.path.join(path, self.nn_model_name)
-        logger.log(15, f'Save Model Hyperparams to {model_path}.')
-        logger.log(15, f'Save Model Text NN weights to {text_nn_path}')
+        model_path = text_nn_path + os.sep + self.model_file_name
```

Comment:
I don't think this is the right way to do saving and loading. Instead do the following (I have tested it works):

```
    def save(self, path: str = None, verbose=True) -> str:
        model = self.model
        self.model = None
        # save this AbstractModel object without NN weights
        path = super().save(path=path, verbose=verbose)
        self.model = model

        text_nn_path = os.path.join(path, self.nn_model_name)
        model.save(text_nn_path)
        logger.log(15, f"\tSaved Text NN weights and model hyperparameters to '{text_nn_path}'.")

        return path
        

    @classmethod
    def load(cls, path: str, reset_paths=True, verbose=True):
        try:
            from autogluon.text.text_prediction.dataset import TabularDataset
            from autogluon.text.text_prediction.models.basic_v1 import BertForTextPredictionBasic
        except ImportError:
            raise ImportError(AG_TEXT_IMPORT_ERROR)

        model = super().load(path=path, reset_paths=reset_paths, verbose=verbose)
        model.model = BertForTextPredictionBasic.load(os.path.join(path, cls.nn_model_name))
        return model

```

As a general rule, all model.pkl files should be located in `{self.path}/model.pkl`.

Reply:
The problem of the original code is that it will be problematic if we are mixing relative path and absolute path. For example:
```python
model_path=./autogluon/SOME_MODEL
path_root_orig=/home/ubuntu/autogluon
```
The previous code will fail to recognize that `./autogluon` is actually equivalent to `/home/ubuntu/autogluon`. Thus, we should use `realpath` to normalize the path data.

Reply:
For the `save()` + `load()` issue, I wasn't aware of this requirement (I think that all we need is consistency). I will change it to `{self.path}/file_name`. 

Also, I noticed that we are using `+` to join the directory and file: https://github.com/awslabs/autogluon/blob/947890e9f48ebd792407517af163d069cb150a27/tabular/src/autogluon/tabular/models/abstract/abstract_model.py#L376

This can be problematic if `path` does not end up with `os.sep`.

Reply:
Please use the code I provided. It does everything correctly. I believe all the issues you are referring to are only present because of the current save/load methods in TextPrediction.

Reply:
Your concern about mixing relative and absolute path is only true if it is possible to happen. The current code makes this impossible. If you still think there is a failure case, please provide a working code snippet showcasing the issue.

Reply:
I used the code that you shared with me initially but then noticed that we are always using `path + file_name` to join directory + filenames. Would you try to read again my comment about the issue (not this one, the one in ensemble model).

Reply:
Okay, you are correct. I haven't seen issues after revising the save + load logic of the text_nn_model in tabular.

Reply:
Could you use my TextModel save/load code instead of the code currently in this PR? It is cleaner, as the current code is saving the model not in `{self.path}/model.pkl` but in `{self.path}/{text_nn_path}/model.pkl`, which we don't want